internal/plugin/connectors/http/generic/config_test.go

```diff
@@ -59,6 +65,32 @@ func Test_newConfig(t *testing.T) {
 		// Assert against value calculated with independent base64 encoder.
 		assert.Equal(t, "Basic Sm9uYWg6c2VjcmV0", headers["Authorization"])
 	})
+
+	t.Run("creates expected query params", func(t *testing.T) {
+		cfg, err := sampleConfig()
+		assert.NoError(t, err)
+		if err != nil {
+			return
+		}
+
+		params, err := renderTemplates(
+			cfg.QueryParams,
+			map[string][]byte{
+				"key":  []byte("someKey"),
+				"name": []byte("Foo Bar"),
+			})
+
+		assert.NoError(t, err)
+		if err != nil {
+			return
+		}
+
+		// A simple test case
+		assert.Equal(t, "Foo Bar", params["NameWithSpaces"])
+
+		// Assert against value calculated with independent base64 encoder.
+		assert.Equal(t, "c29tZUtleQ==", params["Key"])
+	})
```

Comment:
I still don't see a test here that includes special/utf chars. The one that uses `==` isn't adequate enough. See if you can add a test using chars like [these](https://github.com/cyberark/conjur-api-python3/blob/master/test/test_config/policies/variables.yml#L6).